modules/replicator/src/main/java/org/apache/ignite/internal/replicator/Replica.java

```diff
@@ -0,0 +1,161 @@
+import org.apache.ignite.internal.replicator.message.ReadWriteMultiEntryRequest;
+import org.apache.ignite.internal.replicator.message.ReadWriteRequest;
+import org.apache.ignite.internal.replicator.message.ReadWriteSingleEntryRequest;
+import org.apache.ignite.internal.replicator.message.ReplicaMessagesFactory;
+import org.apache.ignite.internal.replicator.message.ReplicaRequest;
+import org.apache.ignite.internal.replicator.message.ReplicaResponse;
+import org.apache.ignite.internal.replicator.message.container.MultiEntryContainer;
+import org.apache.ignite.internal.replicator.message.container.SingleEntryContainer;
+import org.apache.ignite.lang.IgniteInternalException;
+import org.apache.ignite.lang.IgniteStringFormatter;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.network.TopologyService;
+
+/**
+ * Replica server.
+ * TODO:IGNITE-17257 Implement Replica server-side logic.
+ */
+public class Replica implements AutoCloseable {
+    /** Replicator network message factory. */
+    private static final ReplicaMessagesFactory REPLICA_MESSAGES_FACTORY = new ReplicaMessagesFactory();
+
+    /** Replica group identity, this id is the same as the considered partition's id. */
+    private final String partitionId;
+
+    /** Local cluster node. */
+    private final ClusterNode localNode;
+
+    /** Replica listener. */
+    private final ReplicaListener listener;
+
+    /**
+     * The constructor of replica server.
+     *
+     * @param partitionId Identity.
+     * @param topologyService Topology service.
+     * @param listener Replica listener.
+     */
+    public Replica(
+            String partitionId,
+            TopologyService topologyService,
+            ReplicaListener listener
+    ) {
+        this.partitionId = partitionId;
+        this.localNode = topologyService.localMember();
+        this.listener = listener;
+    }
+
+    /**
+     * It is an internal method to applies a RW request to replicator.
+     *
+     * @param req Request to apply in replicator.
+     * @return Response.
+     */
+    private ReplicaResponse applyReadWriteCommandInternal(ReadWriteRequest req) {
+        //TODO: Check locks here.
+        if (req instanceof ReadWriteSingleEntryRequest) {
+            var request = (ReadWriteSingleEntryRequest) req;
+
+            var data = (SingleEntryContainer) listener.applyReadWrite(req.txId(), request.entryContainer(), req.opType());
+
+            return REPLICA_MESSAGES_FACTORY
+                    .singleEntryResponse()
+                    .partitionId(partitionId)
+                    .entryContainer(data)
+                    .build();
+        } else if (req instanceof ReadWriteMultiEntryRequest) {
+            var request = (ReadWriteMultiEntryRequest) req;
+
+            var data = (MultiEntryContainer) listener.applyReadWrite(req.txId(), request.entryContainer(), req.opType());
+
+            return REPLICA_MESSAGES_FACTORY
+                    .multiEntryResponse()
```

Comment:
So that means that given code should be updated, is that correct? Meaning that we should analyze ReplicaResponse inner data and return either mutiEntry or singleEntry message depending on ReplicaResponse attributes.